pkg/syncer/fullsync.go

```diff
@@ -47,6 +47,7 @@ func csiFullSync(ctx context.Context, metadataSyncer *metadataSyncInformer) {
 		log.Errorf("FullSync: Failed to get PVs from kubernetes. Err: %v", err)
 		return
 	}
+	log.Debugf("FullSync: k8sPVs %v", k8sPVs)
```

Comment:
Can you show how log looks like? We should make sure the object is readable in the debug log.
Recently I observed some debug logs that print byteArray instead of a string for some fields in the object. such log will not help.

Reply:
If you have k8s volumes in thousands, this will produce a huge log.

Reply:
The object in the log is readable.

Reply:
```
2020-09-30T22:40:53.155Z        DEBUG   syncer/fullsync.go:50   FullSync: k8sPVs [&PersistentVolume{ObjectMeta:{pvc-0836bc6b-8da7-4133-a94b-eede0b96bf33   /api/v1/persistentvolumes/pvc-0836bc6b-8da7-4133-a94b-eede0b96bf33 24b4ac9e-d1c9-4001-ab03-5314821a9537 274786 0 2020-09-30 22:39:25 +0000 UTC <nil> <nil> map[] map[pv.kubernetes.io/provisioned-by:csi.vsphere.vmware.com] [] [kubernetes.io/pv-protection]  []},Spec:PersistentVolumeSpec{Capacity:ResourceList{storage: {{209715200 0} {<nil>}  BinarySI},},PersistentVolumeSource:PersistentVolumeSource{GCEPersistentDisk:nil,AWSElasticBlockStore:nil,HostPath:nil,Glusterfs:nil,NFS:nil,RBD:nil,ISCSI:nil,Cinder:nil,CephFS:nil,FC:nil,Flocker:nil,FlexVolume:nil,AzureFile:nil,VsphereVolume:nil,Quobyte:nil,AzureDisk:nil,PhotonPersistentDisk:nil,PortworxVolume:nil,ScaleIO:nil,Local:nil,StorageOS:nil,CSI:&CSIPersistentVolumeSource{Driver:csi.vsphere.vmware.com,VolumeHandle:c49f311f-8e69-476f-a1e8-9bca75d62cfc,ReadOnly:false,FSType:ext4,VolumeAttributes:map[string]string{storage.kubernetes.io/csiProvisionerIdentity: 1601504674125-8081-csi.vsphere.vmware.com,type: vSphere CNS Block Volume,},ControllerPublishSecretRef:nil,NodeStageSecretRef:nil,NodePublishSecretRef:nil,ControllerExpandSecretRef:nil,},},AccessModes:[ReadWriteOnce],ClaimRef:&ObjectReference{Kind:PersistentVolumeClaim,Namespace:default,Name:example-vanilla-block-pvc2,UID:0836bc6b-8da7-4133-a94b-eede0b96bf33,APIVersion:v1,ResourceVersion:274774,FieldPath:,},PersistentVolumeReclaimPolicy:Delete,StorageClassName:example-vanilla-block-sc,MountOptions:[],VolumeMode:*Filesystem,NodeAffinity:nil,},Status:PersistentVolumeStatus{Phase:Bound,Message:,Reason:,},} &PersistentVolume{ObjectMeta:{pvc-bfab1faf-04db-4a2a-83b2-1e44a6cdd740   /api/v1/persistentvolumes/pvc-bfab1faf-04db-4a2a-83b2-1e44a6cdd740 61662210-b84d-42eb-a253-35f6ca26bacc 274710 0 2020-09-30 22:39:06 +0000 UTC <nil> <nil> map[] map[pv.kubernetes.io/provisioned-by:csi.vsphere.vmware.com] [] [kubernetes.io/pv-protection]  []},Spec:PersistentVolumeSpec{Capacity:ResourceList{storage: {{209715200 0} {<nil>}  BinarySI},},PersistentVolumeSource:PersistentVolumeSource{GCEPersistentDisk:nil,AWSElasticBlockStore:nil,HostPath:nil,Glusterfs:nil,NFS:nil,RBD:nil,ISCSI:nil,Cinder:nil,CephFS:nil,FC:nil,Flocker:nil,FlexVolume:nil,AzureFile:nil,VsphereVolume:nil,Quobyte:nil,AzureDisk:nil,PhotonPersistentDisk:nil,PortworxVolume:nil,ScaleIO:nil,Local:nil,StorageOS:nil,CSI:&CSIPersistentVolumeSource{Driver:csi.vsphere.vmware.com,VolumeHandle:2f721747-ae4a-4f11-aa47-9dd7a212e84b,ReadOnly:false,FSType:ext4,VolumeAttributes:map[string]string{storage.kubernetes.io/csiProvisionerIdentity: 1601504674125-8081-csi.vsphere.vmware.com,type: vSphere CNS Block Volume,},ControllerPublishSecretRef:nil,NodeStageSecretRef:nil,NodePublishSecretRef:nil,ControllerExpandSecretRef:nil,},},AccessModes:[ReadWriteOnce],ClaimRef:&ObjectReference{Kind:PersistentVolumeClaim,Namespace:default,Name:example-vanilla-block-pvc,UID:bfab1faf-04db-4a2a-83b2-1e44a6cdd740,APIVersion:v1,ResourceVersion:274624,FieldPath:,},PersistentVolumeReclaimPolicy:Delete,StorageClassName:example-vanilla-block-sc,MountOptions:[],VolumeMode:*Filesystem,NodeAffinity:nil,},Status:PersistentVolumeStatus{Phase:Bound,Message:,Reason:,},} &PersistentVolume{ObjectMeta:{pvc-3138fc0f-3998-4425-8ac2-a87049e56651   /api/v1/persistentvolumes/pvc-3138fc0f-3998-4425-8ac2-a87049e56651 dda83323-1301-4402-9e4e-662433867c9d 274764 0 2020-09-30 22:39:22 +0000 UTC <nil> <nil> map[] map[pv.kubernetes.io/provisioned-by:csi.vsphere.vmware.com] [] [kubernetes.io/pv-protection]  []},Spec:PersistentVolumeSpec{Capacity:ResourceList{storage: {{209715200 0} {<nil>}  BinarySI},},PersistentVolumeSource:PersistentVolumeSource{GCEPersistentDisk:nil,AWSElasticBlockStore:nil,HostPath:nil,Glusterfs:nil,NFS:nil,RBD:nil,ISCSI:nil,Cinder:nil,CephFS:nil,FC:nil,Flocker:nil,FlexVolume:nil,AzureFile:nil,VsphereVolume:nil,Quobyte:nil,AzureDisk:nil,PhotonPersistentDisk:nil,PortworxVolume:nil,ScaleIO:nil,Local:nil,StorageOS:nil,CSI:&CSIPersistentVolumeSource{Driver:csi.vsphere.vmware.com,VolumeHandle:a387618e-16b6-4823-a2ec-779be26ce1f0,ReadOnly:false,FSType:ext4,VolumeAttributes:map[string]string{storage.kubernetes.io/csiProvisionerIdentity: 1601504674125-8081-csi.vsphere.vmware.com,type: vSphere CNS Block Volume,},ControllerPublishSecretRef:nil,NodeStageSecretRef:nil,NodePublishSecretRef:nil,ControllerExpandSecretRef:nil,},},AccessModes:[ReadWriteOnce],ClaimRef:&ObjectReference{Kind:PersistentVolumeClaim,Namespace:default,Name:example-vanilla-block-pvc1,UID:3138fc0f-3998-4425-8ac2-a87049e56651,APIVersion:v1,ResourceVersion:274749,FieldPath:,},PersistentVolumeReclaimPolicy:Delete,StorageClassName:example-vanilla-block-sc,MountOptions:[],VolumeMode:*Filesystem,NodeAffinity:nil,},Status:PersistentVolumeStatus{Phase:Bound,Message:,Reason:,},}] {"TraceId": "8ffcb8c7-aae8-477e-8dd3-999d0805e22b"}

```

Reply:
Yes. If we have many volumes, the log will be very huge. Probably we can only log the pv name and status here. @divyenpatel

Reply:
I removed this logs. Since we already has the following debug log in function "getPVsInBoundAvailableOrReleased"  to print out pv name and status.
```
log.Debugf("FullSync: pv %v is in state %v", pv.Name, pv.Status.Phase)
```